backend/pkg/helm/repository.go

```diff
@@ -0,0 +1,261 @@
+package helm
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	zlog "github.com/rs/zerolog/log"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const (
+	defaultNewConfigFileMode   os.FileMode = os.FileMode(0644)
+	defaultNewConfigFolderMode os.FileMode = os.FileMode(0770)
+)
+
+// add repository
+
+type AddUpdateRepoRequest struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+	// TODO: Figure out how to support auth
+	// like username, password, certfile etc
```

Comment:
I looked into this a bit, and it isn't about auth for our helm routes. In this PR the helm routes are currently completely unprotected. Which is obviously a big security issue, since it means any network connection can now run helm commands.

Reply:
This is restricted to desktop. So it's like the dynamic clusters. AFAIU, when trying to apply the charts, it'd fail anyway if no valid credentials to the API server.

Reply:
This auth part is not related to headlamp, it is used when adding a private repository which has its own custom authentication system.

Reply:
To obtain feature parity, we’d need something like this. This is most likely used by internal repos. I asked a mate at one bigco and they have auth on all their internal package repos. This is best practice, so I expect many other places to require this for private repos too.

IMHO we could delay this todo for another time.